src/algebra/regular/basic.lean

```diff
@@ -355,3 +313,15 @@ lemma is_regular_iff_ne_zero [nontrivial R] : is_regular a ↔ a ≠ 0 :=
 ⟨is_regular.ne_zero, is_regular_of_ne_zero⟩
 
 end cancel_monoid_with_zero
+
+/--  Elements of a left cancel semigroup are left regular. -/
+@[to_additive "Elements of an add left cancel semigroup are add-left-regular."]
+lemma is_left_regular_of_left_cancel_semigroup [left_cancel_semigroup R] (g : R) :
```

Comment:
I'd argue these belong back up on the original lines just before we prove something similar for cancel monoids.